src/utils.rs

```diff
@@ -0,0 +1,74 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use tdigest::TDigest;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub enum ValueType {
+    UInt64(u64),
+    F64(f64),
+    String(String),
+    Stats(Stats),
+}
+
+#[derive(Deserialize, Serialize, Default, Debug, Clone, PartialEq)]
+pub struct Stats {
+    pub p99: f64,
+    pub p90: f64,
+    pub mean: f64,
+}
+
+impl Stats {
```

Comment:
Will only saving P99, P90 and mean be flexible enough for people writing rules in Javascript?  Could this be a histogram instead that is lighter weight to traverse than the full data-set but you can reconstitute any stat you wanted?

Reply:
Which stat do you anticipate we would need to have? We can calculate any stat in the backend and have it brought to the UI. The other approach is to have all the data points brought to the front-end and let the user decide which works for them. That will be a rework of the way we are doing things.

Reply:
Yeah. This structure can be updated as needed with any stats we need in the future.